### DISCUSSION TITLE
Introduction to these libs

### DISCUSSION THREAD

**Discussion by aehlke** [open-spaced-repetition/free-spaced-repetition-scheduler#10] [Q&A] [answered]
I'm just starting to learn about FSRS and would like to adopt it in my projects. However I'm looking to clarify some things about the intended usage of these libraries.

The language-specific libs such as the JS or Swift ones look like they contain the scheduler but not the optimizer, correct? Does the optimizer only exist as a Python notebook? To get the optimizer running on platforms such as iOS, is it best to investigate running Python notebooks on iOS or is there an alternative?

Thanks for the help

**Comment by L-M-Sherlock** [comment, marked as answer]
There are two libs including the optimizer: 
- Rust: https://github.com/open-spaced-repetition/fsrs-rs
- Python: https://github.com/open-spaced-repetition/fsrs-optimizer
